src/anomalib/models/uflow/feature_extraction.py

```diff
@@ -0,0 +1,141 @@
+import pytorch_lightning as pl
+import timm
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+AVAILABLE_EXTRACTORS = ['mcait', 'resnet18', 'wide_resnet50_2']
+
+
+def get_feature_extractor(backbone, input_size=256, **kwargs):
+    """
+    Get feature extractor. Currently, is restricted to AVAILABLE_EXTRACTORS.
+    Args:
+        backbone (str): Backbone name.
+        input_size (int): Input size.
+        **kwargs: Additional arguments.
+
+    Returns:
+        FeatureExtractorInterface: Feature extractor.
+    """
+    assert backbone in AVAILABLE_EXTRACTORS, f"Feature extractor must be one of {AVAILABLE_EXTRACTORS}."
+    if backbone in ["resnet18", "wide_resnet50_2"]:
+        return ResNetFeatureExtractor(backbone, input_size, **kwargs)
+    elif backbone == "mcait":
+        return MCaitFeatureExtractor(**kwargs)
+    raise ValueError("`backbone` must be one of `[mcait, resnet18, wide_resnet50_2]`")
+
+
+class FeatureExtractorInterface(pl.LightningModule):
+
+    def __init__(self):
+        super(FeatureExtractorInterface, self).__init__()
+
+    def forward(self, img, **kwargs):
+        raise NotImplementedError
+
+    def extract_features(self, img, **kwargs):
+        raise NotImplementedError
+
+    def normalize_features(self, x, **kwargs):
+        raise NotImplementedError
+
+
+class ResNetFeatureExtractor(FeatureExtractorInterface):
```

Comment:
Would it be possible to use the feature extractor that is already in [here](https://github.com/openvinotoolkit/anomalib/tree/main/src/anomalib/models/components/feature_extractors)?

Reply:
Let me take a look; I think I can integrate it better...